### ISSUE TITLE
"meteor: command not found" in build-meteor-npm-dependencies.sh step

### ISSUE THREAD

**Issue by sverrirsig** [disney/meteor-base#96]
Hi guys,

I've been using this project for version 2.2 of Meteor for some time now without hiccups.

I'm now attempting to update the version to 2.5 and the following step fails during Docker build:

`RUN bash $SCRIPTS_FOLDER/build-meteor-npm-dependencies.sh --build-from-source`

![image](https://user-images.githubusercontent.com/43657200/143265675-c7c2e6d8-90c4-46b9-a984-82df61b08c1c.png)

I'm using example/app-with-native-dependencies.dockerfile with very minor modifications (Just some extra startup scripts etc.)

The only clear difference I could see between the previous template file, and the new 2.5 one, are the base and node images being used.

The feedback loop for trying to figure out exactly what is going wrong is very long since this step happens at the 15 minute mark, I'm also unable to run the build locally due to resource issues so all debugging happens in a CI pipeline 😭 

I just wanted to see if anyone else has hit, and hopefully solved, this before I start diving into the rabbit hole.

Thanks!

**Comment by sverrirsig** [comment]
For future readers, the culprit was that during this step in the build 'meteor npm' was being used opposed to the desired 'npm' command in this script: https://github.com/disney/meteor-base/blob/main/src/docker/build-meteor-npm-dependencies.sh

Past versions have been defaulting to 'npm' for us but after the update _hash npm_ was causing 'meteor npm' to be used instead.

I vaguely recall seeing something about hash being different with newer version of alpine. Although I don't remember the exact details. I won't dive further into this since it seems other people have not been hitting this, and our Dockerfile is somewhat mutated from what the examples show.

TLDR;
Make sure 'npm' is being used in the script opposed to 'meteor npm' - my solution was to just hardcode it and copy the modified script into the scripts folder during the docker build.

**Comment by nscarlson** [comment]
Hardcoding `npm_cmd='npm'` fixed the issue:

```bash
#!/bin/bash

set -o errexit

printf "\n[-] Installing Meteor application server NPM dependencies...\n\n"

npm_cmd='npm'

cd $APP_BUNDLE_FOLDER/bundle/programs/server/
$npm_cmd install

if [[ "$1" = '--build-from-source' ]]; then
	$npm_cmd rebuild --build-from-source
	cd $APP_BUNDLE_FOLDER/bundle/programs/server/npm
	$npm_cmd rebuild --build-from-source
fi
```

I think this would be worth a code change.  This issue cost my entire engineering team many hours.  Need to test this with the non-native dependencies build, however.

**Comment by red-meadow** [comment]
@nscarlson , could you explain what exactly should be done to make this work? Where should I put this modified script?

**Comment by red-meadow** [comment]
@sverrirsig , I think you should reopen this issue. Correct me if I'm wrong, but now it became impossible to use this image as it is, without the hacks described above. I spent many hours on this too.

**Comment by GeoffreyBooth** [comment]
Perhaps a PR would help show the issue and proposed solution?

**Comment by nscarlson** [comment]
@red-meadow The steps you documented there are exactly what I've been doing in the interim.

@GeoffreyBooth Happy to propose a solution in a PR!

**Comment by nscarlson** [comment]
#102

**Comment by s4admin** [comment]
failing for me too...  `meteor not found`

**Comment by GeoffreyBooth** [comment]
@s4admin Can you please provide steps to reproduce? It doesn’t fail in GitHub CI. Perhaps something different about your environment?

**Comment by blacelle** [comment]
We observe this issue with CircleCI (Linux worker) but not with local build (macOS)

**Comment by s4admin** [comment]
> 

Happens with the default sample project on AWS Fargate containers.
